### ISSUE TITLE
config for excluding unused referred var warnings

### ISSUE THREAD

**Issue by borkdude** [borkdude/clj-kondo#364]
e.g.:

```
(ns foo
  (:require
    [dre.util :refer [log]]
    [taoensso.timbre :refer [debug error]]))
```
There has to be a way to suppress `dre.util/log` and `taoensso.timbre/debug + error`

**Comment by borkdude** [comment]
Fixed with https://github.com/borkdude/clj-kondo/commit/b46e90c86cf69a869ae11ca7e00b090932e8d4b8

**Comment by viebel** [comment]
@borkdude 
Still not working for me
```bash
$echo '(ns f (:require [clojure.string :refer [upper-case]])) '  | clj-kondo --config '{:linters {:unused-referred-var {:exclude {clojure.string [upper-case]}}}}' --lint -

<stdin>:1:18: warning: namespace clojure.string is required but never used
linting took 10ms, errors: 0, warnings: 1
$clj-kondo --version
clj-kondo v2019.07.31-alpha
```

**Comment by borkdude** [comment]
@viebel Nice meeting you at Heart of Clojure.

This is the expected behavior. It won't produce a warning for `upper-case`, but here it will for `join`:

```
$ echo '(ns f (:require [clojure.string :refer [upper-case join]])) '  | clj-kondo --config '{:linters {:unused-referred-var {:exclude {clojure.string [upper-case]}}}}' --lint -

<stdin>:1:18: warning: namespace clojure.string is required but never used
<stdin>:1:52: warning: #'clojure.string/join is referred but never used
linting took 8ms, errors: 0, warnings: 2
```

To suppress the warning about `clojure.string` you configure this using the config for `:unused-namespace`.

**Comment by viebel** [comment]
@borkdude It was awesome to meet your for real!
I don't want to configure clojure.string as an allowed unused namespace. I want to allow only unused references to upper-case 
(My real use case is for a debug macro!)
How can I achieve this?

**Comment by borkdude** [comment]
@viebel The warning:

```
<stdin>:1:18: warning: namespace clojure.string is required but never used
```

is coming form the `:unused-namespace` linter, so to suppress this line, you have to configure it like this: `{:linters {:unused-namespace {:exclude [clojure.string]}}}'

The line:

```
<stdin>:1:52: warning: #'clojure.string/join is referred but never used
```
is coming from the `:unused-referred-var` linter, so to suppress that one, you have to use this config:

```
{:linters {:unused-referred-var {:exclude {clojure.string [join]}}}}
```

They are separate linters, so they require separate configuration. Does that make sense?

**Comment by viebel** [comment]
Something is still bothering me.
When I have in my code `(:require [clojure.string :refer [upper-case]])` and I allow `upper-case` to be unreferred, I am expecting clojure.string to be automatically not considered as a unused namespace. The rationale is that the only usage of clojure.string is for `upper-case` which is marked as allowed.

**Comment by borkdude** [comment]
@viebel Right, I guess that makes sense. So when none of the referred vars are considered unused, don't report the namespace as unused. I'll consider that as an improvement for an upcoming release.

**Comment by borkdude** [comment]
@viebel Can you post this in a new issue maybe?
